### ISSUE TITLE
Nodes labels mixed up with Neo4j Output

### ISSUE THREAD

**Issue by lalquier** [knowbi/knowbi-pentaho-pdi-neo4j-output#142]
After the presentation at NODES 2019, I gave this version of kettle a try: kettle-neo4j-remix-8.2.0.7-719-REMIX.zip

I tried it on a simple transformation providing these fields to the Neo4j output step:
Source, Target, Source Label, Target Label, Relationship

This worked fine with the stable plugin from the PDI community edition.
With this version of kettle, the same mapping leads to the creation of nodes with labels that don't make much sense. It feels as if the source and target labels are switched for some reason, which leads to the creation of incorrect nodes.

Is that a known issue with this version?

**Comment by lalquier** [comment]
I did more troubleshooting and I boiled the issue down to the use of Fields for Source/Target node labels and Relationships.
If I use static values for the labels, everything works fine.

The previous version handled label mapping using fields for a single list of mixed relationships.
This version doesn't seem to work the same.

**Comment by mattcasters** [comment]
Thanks for the feedback Laurent. I'll try to find a reproduction path and if I have one I'll add an integration test.

**Comment by lalquier** [comment]
Thanks for the reply. I can provide more details with file samples and screenshots, or even a live call to walk you through what I am doing if you want. Just let me know.
I wasn't sure this is the right place for this level of details (that and attaching pictures to github issues isn't very practical :) ).

**Comment by mattcasters** [comment]
I think I found the culprit in the code.  Will get you a build.

**Comment by mattcasters** [comment]
Could you give release 4.1.2 a try?  Thanks a lot!

**Comment by lalquier** [comment]
Thanks. It will take me a while to find the time to set up maven and compile the code but I will try to do that as soon as I can!

**Comment by mattcasters** [comment]
Or just download a release :+1:

**Comment by mattcasters** [comment]
My colleagues reported the bug as fixed

**Comment by lalquier** [comment]
I finally got around to testing the latest release and I am afraid it is still happening on my end. I did a clean install of Kettle and removed the Neo4j Output plugin before replacing it with the new release (4.1.2). 

I have a test file that adds 12 relationships between nodes that already exist in the database.
After running the import, my number of nodes jump from 1015 to 1021 instead of remaining the same.
